### PR TITLE
Fix incorrect link and add missing ones for pod template

### PR DESCRIPTION
**Pull request by spzala**
The current link to the pod template is pointing to replication controller
page. Also couple places missing a link to get better view of what pod
template is.

> NOTE: Please check the “Allow edits from maintainers” box (see image below) to  
> [allow reviewers to fix problems](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) on your patch and speed up the review process.
>
> Please delete this note before submitting the pull request.

Allow edits from maintainers checkbox

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4366)

### CODE REVIEW

File: docs/concepts/workloads/controllers/daemonset.md [hunk 1]
```diff
@@ -41,9 +41,7 @@ A DaemonSet also needs a [`.spec`](https://git.k8s.io/community/contributors/dev
 
 The `.spec.template` is the only required field of the `.spec`.
 
-The `.spec.template` is a [pod template](/docs/user-guide/replication-controller/#pod-template).
-It has exactly the same schema as a [pod](/docs/user-guide/pods), except
-it is nested and does not have an `apiVersion` or `kind`.
+The `.spec.template` is a [pod template](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview.md#pod-templates). It has exactly the same schema as a [pod](/docs/user-guide/pods), except it is nested and does not have an `apiVersion` or `kind`.
```
Comment (janetkuo) [inline docs/concepts/workloads/controllers/daemonset.md]: `/docs/concepts/workloads/pods/pod-overview/#pod-templates` for the link to pod template

Comment (spzala) [inline docs/concepts/workloads/controllers/daemonset.md]: Thanks @janetkuo I will replace the https link with your suggestion.

**Comment by spzala** [comment]
@janetkuo referencing the links as you suggested seems not working as you could see from my update. Also I noticed there is a problem in this way with many other links - I have opened https://github.com/kubernetes/kubernetes.github.io/issues/4407 and working on it. Please correct me if I am missing anything. Thanks!

**Comment by janetkuo** [comment]
@spzala thanks for the report. Those relative links are supposed to be consumed in https://kubernetes.io/docs/ instead of from GitHub md pages. You can preview your changes in 
deploy/netlify deploy preview:

image

**Comment by spzala** [comment]
@janetkuo ahhh.. thanks for helping understand it. Little tricky but got it I guess :)

**Pull request merged**